core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -360,23 +360,23 @@ private boolean manifestHasDeletedFiles(PartitionAndMetricsEvaluator evaluator,
           dropPartitions.contains(file.specId(), file.partition()) ||
           (isDelete && entry.sequenceNumber() > 0 && entry.sequenceNumber() < minSequenceNumber);
 
-      boolean nonMatchingDeleteFile = !file.content().equals(FileContent.DATA) && !evaluator.rowsMustMatch(file);
```

Comment:
We should avoid extra calls to `evaluator` if possible. Also, there was `isDelete` var already defined.